tests/stubtest_third_party.py

```diff
@@ -97,8 +98,61 @@ def run_stubtest(dist: Path, *, verbose: bool = False, specified_platforms_only:
         if platform_allowlist.exists():
             stubtest_cmd.extend(["--allowlist", str(platform_allowlist)])
 
+        # Perform some black magic in order to run stubtest inside uWSGI
+        # we have to write the exit code from stubtest to a surrogate file
+        # because uwsgi --pyrun does not exit with the exitcode from the
+        # python script.
```

Comment:
I'd definitely prefer it if you could factor out this special-casing into a separate function.

Reply:
Yes, was going to say the same thing

Reply:
Yes, that seems sensible, it's a bit unfortunate that the hack with the exit code can't be factored out quite as nicely. Can you think of a better way? If this solution only has to work on linux/darwin I could create a small sh script which could be run instead of calling uwsgi directly.

Reply:
You are right. That seems like the easiest solution.